clouddq/main.py

```diff
@@ -270,8 +278,26 @@ def main(  # noqa: C901
             gcp_service_account_key_path=gcp_service_account_key_path,
             gcp_impersonation_credentials=gcp_impersonation_credentials,
         )
+        # Load metadata
+        metadata = json.loads(metadata)
         # Set-up cloud logging
-        add_cloud_logging_handler(logger=json_logger)
+        if add_metadata_to_logs:
+            dataplex_batch_name = os.environ.get("DATAPLEX_BATCH_NAME")
+            dataplex_task_name = os.environ.get("DATAPLEX_TASK_NAME")
+            if dataplex_batch_name:
+                metadata["dataplex_batch_name"] = dataplex_batch_name
+            if dataplex_task_name:
+                metadata["dataplex_task_name"] = dataplex_task_name
```

Comment:
can we move this out of the ifelse

Reply:
If we move this out than both the variables will be included into the metadata dictionary. Instead  if add_metadata_to_logs is set to false we need to create a new dictionary for both the variables and pass it into the add_cloud_logging_handler().

Reply:
we can surely move out the part where we are reading the values for the variables.